File: src/main/java/com/akagiyui/drive/model/request/UpdateRoleRequest.java
package com.akagiyui.drive.model.request;

import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.Set;

/**
 * 更新角色 请求
 *
 * @author AkagiYui
 */
@Data
public class UpdateRoleRequest {
    /**
     * 角色名
     */
    @NotNull(message = "{name is missing}")
    private String name;

    /**
     * 角色描述
     */
    private String description;

    /**
     * 角色权限
     */
    private Set<String> permissions;

    /**
     * 是否是默认角色
     */
    private Boolean isDefault;

}

File: app/src/main/kotlin/com/akagiyui/drive/model/request/role/UpdateRoleRequest.kt
package com.akagiyui.drive.model.request.role

import jakarta.validation.constraints.NotNull

/**
 * 更新角色 请求
 *
 * @author AkagiYui
 */
class UpdateRoleRequest {
    /**
     * 角色名
     */
    @NotNull(message = "{name is missing}")
    var name: String? = null

    /**
     * 角色描述
     */
    var description: String? = null

    /**
     * 角色权限
     */
    var permissions: Set<String>? = null

    /**
     * 是否是默认角色
     */
    var isDefault: Boolean? = null
}
